### PR TITLE
chore(identity): add user guide for assigning permissions to applications and introduce concept

### LINKED ISSUE #966: I can read about how to assign permissions to an application
We added the possibility in Identity to assign permissions to applications. This needs to be documented as a user guide and the general concept of permissions of applications needs to be introduced.

Related Epic: https://github.com/camunda-cloud/identity/issues/19
Related Issue: https://github.com/camunda-cloud/identity/issues/650

### PR DESCRIPTION
**Pull request by btepe**
Closes #966

### CODE REVIEW

**Comment by akeller** [comment]
@btepe if this PR is ready for a documentation review, please add @christinaausley as a reviewer. Thanks!

File: docs/self-managed/concepts/access-control/permissions.md [hunk 1]
```diff
@@ -4,7 +4,7 @@ title: "Permissions"
 sidebar_label: "Permissions"
 ---
 
-Permissions allow you to control the level of access a user has to a particular component. Traditionally, this is often described as being able to provide users with "read" or "write" access.
+Permissions allow you to control the level of access a user or an application has to a particular component. Traditionally, this is often described as being able to provide users with "read" or "write" access.
```
Comment (dlavrenuek) [inline docs/self-managed/concepts/access-control/permissions.md:7]:
```suggestion
Permissions allow you to control the level of access a user or an application has to a particular component. Traditionally, this is often described as being able to provide "read" or "write" access.
```

**Comment by christinaausley** [comment]
Happy to review when ready -- feel free to add me when needed.

File: docs/self-managed/identity/user-guide/assigning-a-permission-to-an-application.md [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+
+To assign a permission to an application using Identity, take the following steps:
+
+1. Log in to the Identity UI and navigate to the **Applications** page and click on an application. Then select the **Access to APIs** tab and click on **Assign Permissions**:
+
+![assign-a-permission-application-tab](img/assign-a-permission-application-tab.png)
+
+2. Select the API which contains the permission you want to assign:
+
+![assign-a-permission-application-modal-1](img/assign-a-permission-application-modal-1.png)
```
[the remark this answers is not in the document]
Comment (btepe) [reply docs/self-managed/identity/user-guide/assigning-a-permission-to-an-application.md:17]: That's a good idea 👍 I replaced the screenshots with real Camunda apps and their preset permissions. Let me know what you think.

**Review by dlavrenuek — APPROVED** [review]
Thank you for adjusting the screenshots, looks good from my perspective 👍

**Comment by akeller** [comment]
Is this good to merge @btepe?

**Comment by btepe** [comment]
@christinaausley Thank you for the review! 🙏 @akeller Yes, it is ready from my side!

**Pull request merged**